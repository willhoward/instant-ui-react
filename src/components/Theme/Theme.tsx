import React, { createContext, useContext } from "react";
import { ThemeContextType, ThemeProviderProps } from "./Theme.types";

const defaultTheme = {
  black: "#000000",
};

const ThemeContext = createContext<ThemeContextType | null>(null);

export function ThemeProvider({ children }: ThemeProviderProps) {
  return (
    <ThemeContext.Provider value={defaultTheme}>
      {children}
    </ThemeContext.Provider>
  );
}

// Use this function to access elements of the theme.
export function useTheme() {
  const themeContext = useContext(ThemeContext);

  if (!themeContext) {
    throw new Error("useTheme has to be used within <ThemeProvider>");
  }

  return themeContext;
}
