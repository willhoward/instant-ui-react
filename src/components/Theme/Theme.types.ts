import { ReactNode } from "react";

export interface ThemeContextType {
  black: string;
  darkestGray: string;
  darkGray: string;
  lightGray: string;
  lightestGray: string;
  white: string;
}

export interface ThemeProviderProps {
  children: ReactNode;
}
