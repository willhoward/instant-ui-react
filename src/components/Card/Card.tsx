import React from "react";
import { useTheme } from "../Theme/Theme";
import { CardProps } from "./Card.types";

export function Card({ children = null }: CardProps) {
  const { darkGray } = useTheme();

  const cardStyle = {
    width: "100%",
    height: "auto",
    padding: "24px",
    border: `1px solid ${darkGray}`,
    borderRadius: "12px",
    fontSize: "14px",
    cursor: "pointer",
    background: "transparent",
  };

  return (
    <div style={cardStyle} data-testid="card">
      {children}
    </div>
  );
}
