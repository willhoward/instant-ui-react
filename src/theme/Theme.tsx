import React, { createContext } from "react";
import { ThemeContextType, ThemeProviderProps } from "./Theme.types";

const defaultTheme = {
  colors: {
    black: "#000000",
    darkGray: "#121212",
    midGray: "#232323",
    lightGray: "#A7A7A7",
    white: "#FFFFFF",
  },
};

export const ThemeContext = createContext<ThemeContextType>(defaultTheme);

export const ThemeProvider = ({ children }: ThemeProviderProps) => {
  <ThemeContext.Provider value={defaultTheme}>
    {children}
  </ThemeContext.Provider>;
};
