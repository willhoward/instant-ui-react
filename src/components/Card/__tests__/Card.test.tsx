import React from "react";
import { render } from "@testing-library/react";
import { Card } from "../Card";
import { ThemeProvider } from "../../Theme/Theme";

describe("Card", () => {
  it("Should render child content correctly", () => {
    const { container } = render(
      <ThemeProvider>
        <Card>Example</Card>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
