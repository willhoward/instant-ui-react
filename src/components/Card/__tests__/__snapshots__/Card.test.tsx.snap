// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should render child content correctly 1`] = `
<div>
  <div
    data-testid="card"
    style="width: 100%; height: auto; padding: 24px; border: 1px solid #232323; border-radius: 12px; font-size: 14px; cursor: pointer; background: transparent;"
  >
    Example
  </div>
</div>
`;
